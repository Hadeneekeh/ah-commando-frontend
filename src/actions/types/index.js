export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const MODAL_OPEN = 'MODAL_OPEN';
export const MODAL_CLOSE = 'MODAL_CLOSE';
export const LOADING = 'LOADING';
export const NOT_LOADING = 'NOT_LOADING';
export const GET_PROFILE_START = 'GET_PROFILE_START';
export const GET_PROFILE_SUCCESS = 'GET_PROFILE_SUCCESS';
export const GET_PROFILE_FAILURE = 'GET_PROFILE_FAILURE';

export const GET_ARTICLE_START = 'GET_ARTICLE_START';
export const GET_ARTICLE_SUCCESS = 'GET_ARTICLE_SUCCESS';
export const GET_ARTICLE_FAILURE = 'GET_ARTICLE_FAILURE';

export const EDIT_PROFILE_START = 'EDIT_PROFILE_START';
export const EDIT_PROFILE_SUCCESS = 'EDIT_PROFILE_SUCCESS';
export const EDIT_PROFILE_FAILURE = 'EDIT_PROFILE_FAILURE';

export const POST_IMAGE_START = 'POST_IMAGE_START';
export const POST_IMAGE_SUCCESS = 'POST_IMAGE_SUCCESS';
export const POST_IMAGE_FAILURE = 'POST_IMAGE_FAILURE';

export const UPDATE_FILTER = 'UPDATE_FILTER';
export const DISPLAY_FILTER = 'DISPLAY_FILTER';
export const REMOVE_FILTER = 'REMOVE_FILTER';
export const UPDATE_ARTICLES = 'UPDATE_ARTICLES';
export const UPDATE_SEARCHQUERY = 'UPDATE_SEARCHQUERY';
export const UPDATE_PAGENUMBER = 'UPDATE_PAGENUMBER';
export const CREATE_ARTICLE = 'CREATE_ARTICLE';
export const READ_ARTICLE = 'READ_ARTICLE';

export const UPDATE_CATEGORY_ARTICLES = 'UPDATE_CATEGORY_ARTICLES';
export const UPDATE_CATEGORY = 'UPDATE_CATEGORY';
export const UPDATE_MENU_ITEM = 'UPDATE_MENU_ITEM';
export const UPDATE_CATEGORY_PAGENUMBER = 'UPDATE_CATEGORY_PAGENUMBER';
