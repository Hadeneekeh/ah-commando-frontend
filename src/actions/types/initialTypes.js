export const IS_REDUCING = 'IS_REDUCING';
export const HAS_REDUCED = 'HAS_REDUCED';
