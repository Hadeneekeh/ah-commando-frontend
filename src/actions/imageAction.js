import * as types from '@Actions/types';
import { axiosInstance } from '@Utils/';

export const postImageStart = () => ({
  type: types.POST_IMAGE_START,
});

export const postImageSuccess = (payload) => ({
  type: types.POST_IMAGE_SUCCESS,
  payload,
});

export const postImageFailure = (error) => ({
  type: types.POST_IMAGE_FAILURE,
  error,
});

export const postImage = (payload) => async (dispatch) => {
  dispatch(postImageStart());
  try {
    const response = await axiosInstance.post('/image', payload);
    return dispatch(postImageSuccess(response.data.image));
  } catch (error) {
    return dispatch(postImageFailure(error.response.data));
  }
};
