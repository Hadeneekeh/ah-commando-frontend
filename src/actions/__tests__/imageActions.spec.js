import * as actions from '../imageAction';
import { POST_IMAGE_FAILURE, POST_IMAGE_START } from '../types';

jest.mock('../imageFunc');

describe('Async action creators', () => {
  it('should dispatch post image action', async () => {
    const image = {
      image: 'image.png',
    };
    const dispatch = jest.fn();
    await actions.postImage(image)(dispatch);
    expect(dispatch).toHaveBeenCalledTimes(2);
  });

  it('should dispatch start action on image post', async () => {
    const image = null;
    const dispatch = jest.fn();
    await actions.postImage(image)(dispatch);
    expect(dispatch).toHaveBeenNthCalledWith(1, {
      type: POST_IMAGE_START,
    });
    expect(dispatch).toHaveBeenNthCalledWith(2, {
      type: POST_IMAGE_FAILURE,
      error: {
        error: 'Authorization error',
        status: 401,
      },
    });
  });
});
