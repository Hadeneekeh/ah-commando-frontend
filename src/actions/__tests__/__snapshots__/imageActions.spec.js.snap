// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async action creators Should error as expected 1`] = `
Array [
  Object {
    "type": "POST_IMAGE_START",
  },
  Object {
    "error": Object {
      "response": Object {
        "data": Object {},
      },
    },
    "type": "POST_IMAGE_FAILURE",
  },
]
`;

exports[`Async action creators Should successfully upload an image 1`] = `
Array [
  Object {
    "type": "POST_IMAGE_START",
  },
  Object {
    "payload": undefined,
    "type": "POST_IMAGE_SUCCESS",
  },
]
`;
